velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -559,6 +561,77 @@ TpchPlan TpchQueryBuilder::getQ10Plan() const {
   return context;
 }
 
+TpchPlan TpchQueryBuilder::getQ12Plan() const {
+  std::vector<std::string> ordersColumns = {"o_orderkey", "o_orderpriority"};
+  std::vector<std::string> lineitemColumns = {
+      "l_receiptdate",
+      "l_orderkey",
+      "l_commitdate",
+      "l_shipmode",
+      "l_shipdate"};
+
+  auto ordersSelectedRowType = getRowType(kOrders, ordersColumns);
+  const auto& ordersFileColumns = getFileColumnNames(kOrders);
+  auto lineitemSelectedRowType = getRowType(kLineitem, lineitemColumns);
+  const auto& lineitemFileColumns = getFileColumnNames(kLineitem);
+
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  core::PlanNodeId ordersScanNodeId;
+  core::PlanNodeId lineitemScanNodeId;
+
+  auto receiptDate = "l_receiptdate";
+  std::string receiptDateFilter = formatDateFilter(
+      receiptDate, lineitemSelectedRowType, "'1994-01-01'", "'1994-12-31'");
+  std::string commitDateFilter = "l_commitdate < l_receiptdate";
```

Comment:
Thanks for sharing, Presto is not doing this optimization, it just applies the following filter:
((receiptdate) >= (DATE'1994-01-01'))) AND (((receiptdate) < (DATE'1995-01-01')) AND ((commitdate) < (receiptdate)))) AND ((shipdate) < (commitdate))
I will apply this optimization and recompute the benchmark values

Reply:
just to clarify, adding a subfield filter `l_commitdata <= 1994-12-31` for the table scan should do.

Reply:
This is a good idea. Lets try this predicate. If the tableScan becomes way more expensive like we have been seeing elsewhere, this will be a good use-case for us to motivate perf improvements there.

Reply:
@pramodsatya can you share any insights in the performance after adding this predicate.

Reply:
Hi Deepak, the numbers in PR description are obtained after adding the predicate, for 1 driver, we see an improvement from 9.84s to 8.63s, for 16 drivers the time improved from 1.99s to 1.93s, I will add the earlier numbers from version history into a separate slide